core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1551,7 +1551,7 @@ class KafkaApis(val requestChannel: RequestChannel,
 
       val (existingAndAuthorizedForDescribeTopics, nonExistingOrUnauthorizedForDescribeTopics) =
         partitionsToAdd.asScala.partition { tp =>
-          authorize(request.session, Describe, new Resource(Topic, tp.topic)) && metadataCache.contains(tp.topic)
+          authorize(request.session, Describe, new Resource(Topic, tp.topic)) && metadataCache.contains(tp.topic, tp.partition)
```

Comment:
That's a good question: I think it will only be problematic if the topic does exist but not with the specified partitionId, e.g. for internal topic we do not need to check for partition since they should never change, and for `produceRequest` if it is sending to `topicA, partition8` while `topicA` only have 8 partitions, this would be a problem. But since `partition8` is calculated on the producer side's partitioner it is usually safe as topic cannot shrink partitions today.

I think to be safer we can change them to check on both topic / partition, maybe worth a separate PR?